### PR TITLE
Create nft

### PR DESCRIPTION
**Pull request by haybankz**
### Description
adds creation of NFT from data inputted by user

I have
- [ ] implemented upload of NFT image on nft.storage
- [ ] implemented share functionality (Easel generated URL)

### CODE REVIEW

File: lib/screens/home_screen.dart [hunk 1]
```diff
@@ -101,31 +103,29 @@ class _HomeScreenState extends State<HomeScreen> {
...
+                                  size: 18,
+                                ),
+                                icon: Text(
+                                  "Mint more",
+                                  style: Theme.of(context)
+                                      .textTheme
+                                      .bodyText1!
+                                      .copyWith(
+                                          fontSize: 20,
+                                          color: const Color(0xFF1212C4),
```
Comment (kjawadDeveloper1) [inline lib/screens/home_screen.dart]: Centralize the colors please.

File: lib/screens/home_screen.dart [hunk 2]
```diff
@@ -101,31 +103,29 @@ class _HomeScreenState extends State<HomeScreen> {
...
+                      builder: (_, int currentPage, __) => _currentPage.value == 3
+                          ? Consumer<EaselProvider>(
+                            builder: (_, provider, __) => TextButton.icon(
+                                onPressed: () {
+                                  provider.initStore();
+                                  _pageController.jumpToPage(0);
+                                },
+                                label: const Icon(
+                                  Icons.arrow_forward_ios,
+                                  color: Color(0xFF1212C4),
```
Comment (kjawadDeveloper1) [inline lib/screens/home_screen.dart]: Centralize the colors.

**Review by kjawadDeveloper1 — CHANGES_REQUESTED** [review]

File: lib/screens/description_screen.dart [hunk 3]
```diff
@@ -101,10 +114,11 @@ class _DescriptionScreenState extends State<DescriptionScreen> {
...
                         ],
                         controller: provider.noOfEditionController,
-                        validator: (value) {
-                          if (value!.isEmpty) return "Number of Editions";
-                          if (int.parse(value) > 10000)
-                            return "Maximum is 10,000";
+                        validator: (value){
+                          if(value!.isEmpty) return "Enter number of editions";
+                          if(int.parse(value) == 0) return "Minimum is 1";
+                          if(int.parse(value) > 10000) return "Maximum is 10,000";
```
Comment (mustafapylons) [inline lib/screens/description_screen.dart]: Let's not use Magic numbers in the code (like 10000, use constants etc.)

File: lib/screens/description_screen.dart [hunk 4]
```diff
@@ -73,13 +73,11 @@ class _DescriptionScreenState extends State<DescriptionScreen> {
...
-                        ],
-                        validator: (value) {
-                          if (value!.isEmpty) return "Enter NFT description";
-                          if (value.length < 10)
-                            return "Enter at least 10 characters";
+
+                        inputFormatters: [LengthLimitingTextInputFormatter(256)],
+                        validator: (value){
+                          if(value!.isEmpty) return "Enter NFT description";
+                          if(value.length <= 20) return "Enter more than 20 characters";
```
Comment (mustafapylons) [inline lib/screens/description_screen.dart]: Same here, maximum values, boundaries etc. should be in constant variables.

File: lib/utils/file_utils.dart [hunk 5]
```diff
@@ -35,7 +50,63 @@ class FileUtils {
...
+
+    // Convert to ui.Picture
+    ui.Picture picture = svgDrawableRoot.toPicture(size: Size(width, height));
+
+    // Convert to ui.Image. toImage() takes width and height as parameters
+    // you need to find the best size to suit your needs and take into account the screen DPI
+    ui.Image image = await picture.toImage(width.toInt(), height.toInt());
+    ByteData? bytes = await image.toByteData(format: ui.ImageByteFormat.png);
+
+    // finally to Bitmap
```
Comment (mustafapylons) [inline lib/utils/file_utils.dart]: Dead Code in multiple places, please remove dead code.

File: lib/datasources/remote_datasource.dart [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    return _dio.post("/upload", data: Stream.fromIterable(file.readAsBytesSync().map((e) => [e]))).then((response){
+      if(response.statusCode == 200){
+        final data =  StorageResponseModel.fromJson(response.data);
+        return ApiResponse<StorageResponseModel>.success(data: data);
+      }
+
+      return ApiResponse<StorageResponseModel>.error(errorMessage: response.data["error"]["message"] ?? "Upload Failed");
+
+    }).catchError((e) => ApiResponse<StorageResponseModel>.error(errorMessage: e.toString()));
```
Comment (mustafapylons) [inline lib/datasources/remote_datasource.dart]: Use meaningful variables, not "e".

**Review by mustafapylons — COMMENTED** [review]
Left comments, please correct them before approval.

**Comment by haybankz** [comment]
@kjawadDeveloper1 @mustafapylons review issues have been resolved

**Review by mustafapylons — APPROVED** [review]

**Review by starhelios — APPROVED** [review]

**Review by kjawadDeveloper2 — APPROVED** [review]
Looking Good from my side.

**Pull request merged**